src/main/webapp/jsp/studentCourseJoinConfirmation.jsp

```diff
@@ -15,18 +15,47 @@
 	    </div>
 	    <div class="panel-body">
 	        <p>
-	            You are currently logged in as <span><strong>${data.account.googleId}</strong></span>. 
-	            <br>If this is not you please <a href="${data.logoutUrl}">log out</a> and re-login using your own Google account.
-	            <br>If this is you, please confirm below to complete your registration.
+	            You are currently logged in as <span><strong>${data.account.googleId}</strong></span>.
+	            <c:if test="${data.redirectResult}">
+	                You have been redirected to this page because you navigated to a link for the course <strong>${data.courseId}</strong>, which you have not been registered in.
+	            </c:if>
 	            <br>
+	            <c:choose>
+	                <c:when test="${data.redirectResult}">
+	                    If you wish to register as <strong>${data.account.googleId}</strong>, please confirm below to complete your registration.
+	                    <br>If you wish to register with another Google ID or do not wish to register for this course, please
+	                    <a href="${data.logoutUrl}">log out</a>
+	                    , log in with your desired Google ID if necessary, and navigate to the link sent to you via email again.
```

Comment:
Now that we are using this message whenever the next_url is a page that requires registration, we might have to update the message here for future-proofing. Especially the part about "navigating to the link that came via email" needs to be updated as now the user may not be allowed to access the next_url without registering at all...